### PR TITLE
Cache driver name

### PR DESCRIPTION
**Pull request by jsafrane**
Provisioner should not ask for it every time, it should not change during driver container lifetime.

### CODE REVIEW

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -339,14 +336,15 @@ func NewCSIProvisioner(client kubernetes.Interface,
...
+		driverName:           driverName,
 	}
 	return provisioner
 }
 
 // This function get called before any attempt to communicate with the driver.
 // Before initiating Create/Delete API calls provisioner checks if Capabilities:
 // PluginControllerService,  ControllerCreateVolume sre supported and gets the  driver name.
-func checkDriverState(grpcClient *grpc.ClientConn, timeout time.Duration, needSnapshotSupport bool) (*driverState, error) {
+func checkDriverCapabilities(grpcClient *grpc.ClientConn, timeout time.Duration, needSnapshotSupport bool) (sets.Int, error) {
```
Comment (davidz627) [inline pkg/controller/controller.go]: Why are we representing capabilities with `sets.Int`, is there no named type from the protobuf we can use? if not, can we define one
`type PluginCapabilitiesType sets.Int`

File: cmd/csi-provisioner/csi-provisioner.go [hunk 2]
```diff
@@ -145,7 +144,7 @@ func init() {
 
 	// Create the provisioner: it implements the Provisioner interface expected by
 	// the controller
-	csiProvisioner := ctrl.NewCSIProvisioner(clientset, csiAPIClient, *csiEndpoint, *connectionTimeout, identity, *volumeNamePrefix, *volumeNameUUIDLength, grpcClient, snapClient)
+	csiProvisioner := ctrl.NewCSIProvisioner(clientset, csiAPIClient, *csiEndpoint, *connectionTimeout, identity, *volumeNamePrefix, *volumeNameUUIDLength, grpcClient, snapClient, provisionerName)
```
Comment (davidz627) [inline cmd/csi-provisioner/csi-provisioner.go]: `provisionerName == driverName`? why do we even have these two concepts then. Can we just call it `driverName` everywhere?

Comment (jsafrane) [reply cmd/csi-provisioner/csi-provisioner.go]: I think it's mandatory for volume topology, where kubelet knows just driver name from driver registrar and scheduler knows only provider name from StorageClass. @msau42 knows details.

File: pkg/controller/controller.go (hunk 1 above)
Comment (jsafrane) [reply pkg/controller/controller.go]: I tried, but then I loose all `sets.Int` methods (no `PluginCapabilitiesType.Has`). I could restore `driverState` struct (with a better name) with `sets.Int` as a field, but that looks ugly.

File: cmd/csi-provisioner/csi-provisioner.go (hunk 2 above)
Comment (msau42) [reply cmd/csi-provisioner/csi-provisioner.go]: Yes, they're the same.  external-provisioner originally was implemented allowing them to be different but there's no reason they need to be so we removed it in https://github.com/kubernetes-csi/external-provisioner/pull/186

**Pull request merged**